core/src/main/scala/zio/logging/Logging.scala

```diff
@@ -1,21 +1,72 @@
 package zio.logging
 
-import zio.{ Cause, ZIO }
+import zio.clock.{ currentDateTime, Clock }
+import zio.console.{ putStrLn, Console }
+import zio.logging.Logger.classNameForLambda
+import zio.{ UIO, URIO, ZIO }
+
+trait Logging[-R] {
+
+  def logger: Logger[R]
 
-trait Logging[Message] {
-  def logging: Logging.Service[Any, Message]
 }
 
 object Logging {
 
-  trait Service[-R, Message] {
+  def make[R](logger: (LogContext, => String) => URIO[R, Unit]) =
+    Logger
+      .make(logger)
+      .map(l =>
+        new Logging[R] {
+          override def logger: Logger[R] = l
+        }
+      )
+
+  val ignore: UIO[Logging[Any]] =
+    make((_, _) => ZIO.unit)
+
+  def console(
+    format: (LogContext, => String) => String
+  ): URIO[Console with Clock, Logging[Console with Clock]] =
+    make((context, line) =>
+      for {
+        date  <- currentDateTime
+        level = context.get(LogAnnotation.Level)
+        loggerName = context.get(LogAnnotation.Name) match {
+          case Nil   => classNameForLambda(line).getOrElse("ZIO.defaultLogger")
+          case names => LogAnnotation.Name.render(names)
+        }
+        _ <- putStrLn(date.toString + " " + level.render + " " + loggerName + " " + format(context, line))
+      } yield ()
+    )
+
+  def log(line: => String): ZIO[Logging[Any], Nothing, Unit] =
+    ZIO.accessM[Logging[Any]](_.logger.log(line))
+
+  def log(level: LogLevel)(line: => String): ZIO[Logging[Any], Nothing, Unit] =
+    ZIO.accessM[Logging[Any]](_.logger.log(level)(line))
+
+  def locallyAnnotate[A, R <: Logging[Any], E, A1](annotation: LogAnnotation[A], value: A)(
+    zio: ZIO[R, E, A1]
+  ): ZIO[Logging[Any] with R, E, A1] =
+    ZIO.accessM[Logging[Any] with R](_.logger.locallyAnnotate[A, R, E, A1](annotation, value)(zio))
+
+  def contramap[A1](f: A1 => String): URIO[Logging[Any], LoggerLike[A1, Any]] =
```

Comment:
This doesn't feel right here. I think I'd delete it.

Reply:
I had the same feeling. if some wanna do it could still use ```logger``` method and then ```contramap```